main.c

```diff
@@ -129,8 +136,8 @@ int main(int argc, char *argv[]) {
 		save_todos(savedetail);
 	} else if (strcmp(action, "finish") == 0 || strcmp(action, "done") == 0) {
 		if (argc != 2) panic("Argument incomplete.\n");
-		char *tmp;
-		long long index = strtoll(argv[1], &tmp, 10);
+		long long index = strtoll(argv[1], NULL, 10);
```

Comment:
I am guessing that ID 0 is not valid in your application?

If not... then the check how it failed would be nice, otherwise it might be ok.

Reply:
User use `1` as the first ID. Meanwhile, the program use `0` as the first id as the next line confirms it with decreasing the number by 1.